libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -855,11 +855,33 @@ void AC_PosControl::input_vel_accel_z(float &vel, const float accel, bool ignore
 
 /// set_pos_target_z_from_climb_rate_cm - adjusts target up or down using a commanded climb rate in cm/s
 ///     using the default position control kinematic path.
+///     The zero target altitude is varied to follow pos_offset_z
+void AC_PosControl::set_pos_target_z_from_climb_rate_cm(const float vel, float pos_offset_z)
+{
+    // remove terrain offsets for flat earth assumption
+    _pos_target.z -= _pos_offset_z;
+    _vel_desired.z -= _vel_offset_z;
+    _accel_desired.z -= _accel_offset_z;
+
+    float vel_temp = vel;
+    input_vel_accel_z(vel_temp, 0, false);
+
+    // update the vertical position, velocity and acceleration offsets
+    update_pos_offset_z(pos_offset_z);
+
+    // add terrain offsets
+    _pos_target.z += _pos_offset_z;
+    _vel_desired.z += _vel_offset_z;
+    _accel_desired.z += _accel_offset_z;
+}
+
+/// set_land_from_climb_rate_cm - adjusts target up or down using a commanded climb rate in cm/s
+///     using the default position control kinematic path.
 ///     ignore_descent_limit turns off output saturation handling to aid in landing detection. ignore_descent_limit should be true unless landing.
-void AC_PosControl::set_pos_target_z_from_climb_rate_cm(const float vel, bool ignore_descent_limit)
+void AC_PosControl::set_land_from_climb_rate_cm(const float vel, bool ignore_descent_limit)
```

Comment:
maybe we should call, "set_land_from_climb_rate_cm", "land_at_climb_rate_cm".

Reply:
Maybe we should adjust the comments so that it is clear when this should be used vs set_pos_target_z_from_climb_rate_cm?  .. and if should always be used for landing then I guess we would expect the "vel" parameter to always be negative?  maybe we should clarify this in the comments as well.